cirq/ops/raw_types.py

```diff
@@ -578,8 +578,13 @@ def _resolve_parameters_(self, resolver):
 
     def _circuit_diagram_info_(self, args: 'cirq.CircuitDiagramInfoArgs'
                               ) -> 'cirq.CircuitDiagramInfo':
-        return protocols.circuit_diagram_info(self.sub_operation, args,
-                                              NotImplemented)
+        sub_op_info = protocols.circuit_diagram_info(self.sub_operation, args,
+                                                     NotImplemented)
+        if args and args.include_tags:
+            sub_op_info.wire_symbols = (
+                (sub_op_info.wire_symbols[0] + str(list(self._tags)),) +
```

Comment:
Yup.  Fixed this and added a test, though it seems that returning NotImplemented returns a TypeError from the protocol

Reply:
`args` will never be None.  The protocol handles the default value for `args`.

If `_circuit_diagram_info_` returns `NotImplemented`, the diagram drawer will fallback to the gate repr.  With this code, tags won't be printed for gates that don't implement `_circuit_diagram_info_`.  I've had a similar issue dealing with `NotImplemented` in the past when wrapping gates.  I think the long-term solution is to make `cirq.circuit_diagram_info` handle the default case instead of the diagram drawer and always return a valid `CircuitDiagramInfo` object.

Reply:
I added some code to handle TaggedOperations within the drawer classes in circuit.  I was a little hesitant to directly modify the protocol code.

I also added a test for a gate without a _circuit_diagram_info_

What do you think?